common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -663,6 +661,11 @@ def _compute_metadata_inheritance_tree(self, course_id):
             # manually pick it apart b/c the db has tag and we want as_published revision regardless
             location = as_published(Location._from_deprecated_son(result['_id'], course_id.run))
 
+            # TODO This level of detail should not be here. It is necessary because the display name for video
+            # modules defaults to "Video" if one is not provided. We should find a better method of setting this value.
+            if result['_id']['category'] == 'video' and not result['metadata'].get('display_name'):
+                result['metadata']['display_name'] = 'Video'
```

Comment:
Actually, in this particular case of videos not having names (because of a data issue), maybe it's better for the thing calling this to handle it properly (e.g. just show the enclosing vertical's name if the video is unnamed).